diff --git a/src/binary64/cosh/cosh.c b/src/binary64/cosh/cosh.c
index b5e31cba..e97f623f 100644
--- a/src/binary64/cosh/cosh.c
+++ b/src/binary64/cosh/cosh.c
@@ -104,7 +104,7 @@ static double __attribute__((noinline)) as_cosh_zero(double x){
   double y0 = fasttwosum(1.0, y1, &y1);
   y1 = fasttwosum(y1, y2, &y2);
   b64u64_u t = {.f = y1};
-  if(__builtin_expect(!(t.u&(~0ul>>12)), 0)){
+  if(__builtin_expect(!(t.u&(~0ull>>12)), 0)){
     b64u64_u w = {.f = y2};
     if((w.u^t.u)>>63)
       t.u--;
@@ -112,7 +112,7 @@ static double __attribute__((noinline)) as_cosh_zero(double x){
       t.u++;
     y1 = t.f;
   }
-  if(__builtin_expect((t.u&(~0ul>>12))==(~0ul>>12), 0)) return as_cosh_database(x, y0 + y1);
+  if(__builtin_expect((t.u&(~0ull>>12))==(~0ull>>12), 0)) return as_cosh_database(x, y0 + y1);
   return y0 + y1;
 }
 
@@ -311,7 +311,7 @@ double cr_cosh(double x){
       th = as_exp_accurate(ax, t, th, tl, &tl);
       th = fasttwosum(th, tl, &tl);
       b64u64_u uh = {.f = th}, ul = {.f = tl};
-      int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ul>>12);
+      int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ull>>12);
       th += tl;
       th *= 2;
       th *= sp.f;
@@ -371,7 +371,7 @@ double cr_cosh(double x){
   }
   rh = fasttwosum(rh, rl, &rl);
   b64u64_u uh = {.f = rh}, ul = {.f = rl};
-  int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ul>>12);
+  int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ull>>12);
   rh += rl;
   if(__builtin_expect(ml<=16 || eh-el>103,0)) return as_cosh_database(x, rh);
   return rh;
diff --git a/src/binary64/sinh/sinh.c b/src/binary64/sinh/sinh.c
index e9b24d45..98526d40 100644
--- a/src/binary64/sinh/sinh.c
+++ b/src/binary64/sinh/sinh.c
@@ -110,7 +110,7 @@ static double __attribute__((noinline)) as_sinh_zero(double x){
   double y0 = fasttwosum(x, y1, &y1);
   y1 = fasttwosum(y1, y2, &y2);
   b64u64_u t = {.f = y1};
-  if(__builtin_expect(!(t.u&(~0ul>>12)), 0)){
+  if(__builtin_expect(!(t.u&(~0ull>>12)), 0)){
     b64u64_u w = {.f = y2};
     if((w.u^t.u)>>63)
       t.u--;
@@ -350,7 +350,7 @@ double cr_sinh(double x){
       th *= __builtin_copysign(1, x);
       tl *= __builtin_copysign(1, x);
       b64u64_u uh = {.f = th}, ul = {.f = tl};
-      int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ul>>12);
+      int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ull>>12);
       th += tl;
       th *= 2;
       th *= sp.f;
@@ -410,7 +410,7 @@ double cr_sinh(double x){
   }
   rh = fasttwosum(rh, rl, &rl);
   b64u64_u uh = {.f = rh}, ul = {.f = rl};
-  int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ul>>12);
+  int64_t eh = (uh.u>>52)&0x7ff, el = (ul.u>>52)&0x7ff, ml = (ul.u + 8)&(~0ull>>12);
   rh *= __builtin_copysign(1, x);
   rl *= __builtin_copysign(1, x);
   rh += rl;
diff --git a/src/binary64/tanh/tanh.c b/src/binary64/tanh/tanh.c
index 9aea80b6..6b7ae8ee 100644
--- a/src/binary64/tanh/tanh.c
+++ b/src/binary64/tanh/tanh.c
@@ -133,7 +133,7 @@ static double __attribute__((noinline)) as_tanh_zero(double x){ // |x|<0.25
     double y0 = fasttwosum(x, y1, &y1);
     y1 = fasttwosum(y1, y2, &y2);
     b64u64_u t = {.f = y1};
-    if (__builtin_expect(!(t.u & (~0ul >> 12)), 0)) {
+    if (__builtin_expect(!(t.u & (~0ull >> 12)), 0)) {
         b64u64_u w = {.f = y2};
         if ((w.u ^ t.u) >> 63)
             t.u--;
@@ -382,6 +382,6 @@ double cr_tanh(double x){
   rh = fasttwosum(rh, rl, &rl);
   double res = __builtin_copysign(2,x)*rh + __builtin_copysign(2,x)*rl;
   b64u64_u lu = {.f = rl};
-  if(__builtin_expect(((lu.u+32)&(~0ul>>12))<65, 0)) return as_tanh_database(x, res);
+  if(__builtin_expect(((lu.u+32)&(~0ull>>12))<65, 0)) return as_tanh_database(x, res);
   return res;
 }
